pkg/drivers/powerstore.go

```diff
@@ -61,6 +61,10 @@ func PrecheckPowerStore(ctx context.Context, cr *csmv1.ContainerStorageModule, o
 	// Check for secret only
 	config := cr.Name + "-config"
 
+	if cr.Spec.Driver.AuthSecret != "" {
+		config = cr.Spec.Driver.AuthSecret
+	}
+
```

Comment:
@karthikk92, please check if the comments above AuthSecret field in all powerstore samples still holds good with this fix